### PR TITLE
Let the project to use the channel's color as default when it is set

### PR DESCRIPTION
**Pull request by Maychell**
When the project is linked to a channel, it should use the channel's main color as default. If it's empty or the project has no channel, use the project's color instead.

### CODE REVIEW

File: spec/models/project_spec.rb [hunk 1]
```diff
@@ -1937,25 +1937,50 @@
...
+        context "when the project has category" do
+          let(:project)        { create(:project, channels: [channel], category: category) }
+          let(:channel)        { create(:channel, :non_recurring, main_color: '') }
+          let(:category)       { create(:category, color: category_color) }
+          let(:category_color) { 'c_color' }
 
-      it "should return the default color" do
-        is_expected.to eq(default_color)
+          it "returns the category's color" do
+            expect(project.color).to eq category_color
```
Comment (jonatasrancan) [inline spec/models/project_spec.rb]: Just a small detail.
I think you don't need `let(:category_color)...`, you can do it straight forward by just adding the color direct in the category, and in the test just call `category.color`. By doing this you, when you read the test, you enforce that the color comes from the `category`.
What do you think?
```ruby
....
let(:category) { create(:category, color: 'c_color' }

 it "returns the category's color" do
   expect(project.color).to eq category.color
.....
````


Same comment about the `main_color` in the `channel`

File: app/models/project.rb [hunk 2]
```diff
@@ -310,7 +310,7 @@ def recurring?
   end
 
   def color
-    recurring? ? CatarseSettings[:default_color] : category.color
+    @color ||= channel_color || category_color || CatarseSettings[:default_color]
```
Comment (jonatasrancan) [inline app/models/project.rb]: Another question: Do you guys use decorators in the project?
If yes, could this method be moved to the decorator?

Comment (Maychell) [inline app/models/project.rb]: We do have. Yeah, you are right, I'll move it to the decorator. Thanks :)

File: spec/decorators/project_decorator_spec.rb [hunk 3]
```diff
@@ -369,4 +369,51 @@
       end
     end
   end
+
+  describe "#color" do
+    before { CatarseSettings[:default_color] = '#ff8a41' }
+
+    context "when the project is in a channel" do
```
Comment (igorrfc) [inline spec/decorators/project_decorator_spec.rb]: `when the project is within a channel` :nail_care:

**Pull request merged**